internal/restorer/restorer_unix_test.go

```diff
@@ -59,3 +60,97 @@ func TestRestorerRestoreEmptyHardlinkedFileds(t *testing.T) {
 		rtest.Equals(t, s1.Ino, s2.Ino)
 	}
 }
+
+func checkConsistentInfo(t testing.TB, file string, fi os.FileInfo, modtime time.Time, mode os.FileMode) {
+	if fi.Mode() != mode {
+		t.Errorf("checking %q, Mode() returned wrong value, want 0%o, got 0%o", file, mode, fi.Mode())
+	}
+
+	if !fi.ModTime().Equal(modtime) {
+		t.Errorf("checking %s, ModTime() returned wrong value, want %v, got %v", file, modtime, fi.ModTime())
+	}
+}
+
+// test inspired from test case https://github.com/restic/restic/issues/1212
+func TestRestorerConsistentTimestampsAndPermissions(t *testing.T) {
+	timeForTest := time.Date(2019, time.January, 9, 1, 46, 40, 0, time.UTC)
+
+	repo, cleanup := repository.TestRepository(t)
+	defer cleanup()
+
+	_, id := saveSnapshot(t, repo, Snapshot{
+		Nodes: map[string]Node{
+			"dir": Dir{
+				Mode:    0750 | os.ModeDir,
+				ModTime: timeForTest,
+				Nodes: map[string]Node{
+					"file1": File{
+						Mode:    os.FileMode(0700),
+						ModTime: timeForTest,
+						Data:    "content: file\n",
+					},
+					"anotherfile": File{
+						Data: "content: file\n",
+					},
+					"subdir": Dir{
+						Mode:    0700 | os.ModeDir,
+						ModTime: timeForTest,
+						Nodes: map[string]Node{
+							"file2": File{
+								Mode:    os.FileMode(0666),
+								ModTime: timeForTest,
+								Links:   2,
+								Inode:   1,
+							},
+						},
+					},
+				},
+			},
+		},
+	})
+
+	res, err := NewRestorer(repo, id)
+	rtest.OK(t, err)
+
+	res.SelectFilter = func(item string, dstpath string, node *restic.Node) (selectedForRestore bool, childMayBeSelected bool) {
+		switch item {
```

Comment:
please use `filepath.ToSlash(item)` here, then the switch statement can be identical across operating systems.

Reply:
Done.